install.sh

```diff
@@ -615,5 +685,76 @@ fi
 
 }
 
+function register_sandcats_name() {
+  # We allow environment variables to override some details of the
+  # Sandcats service, so that during development, we can test against
+  # a non-production Sandcats service.
+  SANDCATS_BASE_DOMAIN="${OVERRIDE_SANDCATS_BASE_DOMAIN:-sandcats.io}"
+  SANDCATS_API_BASE="${OVERRIDE_SANDCATS_API_BASE:-https://sandcats.io}"
+  SANDCATS_CURL_PARAMS="${OVERRIDE_SANDCATS_CURL_PARAMS:-}"
+
+  echo "Press enter to accept a random name, or choose your own (alphanumeric, max len 20)."
+  echo "Type the word none to skip this step."
+  DESIRED_SANDCATS_NAME=$(prompt "What *.${SANDCATS_BASE_DOMAIN} subdomain would you like?" '')
+  if [ -z "$DESIRED_SANDCATS_NAME" ] ; then
+    DESIRED_SANDCATS_NAME=$(random_sandcats_name)
+  fi
+
+  # If the user really wants none of our sandcats help, then bail out.
+  if [ "none" = "$DESIRED_SANDCATS_NAME" ] ; then
+    return
+  fi
+
+  # Validate the client-side, to avoid problems, against the same
+  # regex that the server is using.
+  [[ $DESIRED_SANDCATS_NAME =~ ^[0-9a-zA-Z]{1,20}$ ]] || register_sandcats_name
```

Comment:
Also does the recursion on this line actually work? If the `register_sandcats_name` recursion returns won't we incorrectly go on and repeat the rest of the function? (I.e. this isn't _tail_ recursion here...)